src/Compilers/Core/Portable/SourceFileResolver.cs

```diff
@@ -39,21 +48,62 @@ public SourceFileResolver(ImmutableArray<string> searchPaths, string baseDirecto
 
             _baseDirectory = baseDirectory;
             _searchPaths = searchPaths;
-        }
+            _pathMap = pathMap;
 
-        public string BaseDirectory
-        {
-            get { return _baseDirectory; }
+            // the keys in pathMap should not end with a path separator
+            if (!pathMap.IsDefaultOrEmpty) foreach (var kv in pathMap)
+            {
+                var key = kv.Key;
+                if (key == null || key.Length == 0)
+                {
+                    throw new ArgumentException("empty key", nameof(pathMap));
```

Comment:
Yes. We might miss some, but we pretty much localize all of them using current culture. Think about this particular case. The string that you pass to the constructor will be injected into a "bigger" message, which is very likely to be localized. If we don't localize our part, the whole message will look ugly.